src/i965_encoder_vp8.c

```diff
@@ -0,0 +1,6239 @@
+            vp8_context->down_scaled_height_in_mb16x = vp8_context->min_scaled_dimension_in_mbs;
+        }
+    }
+}
+
+static void
+i965_encoder_vp8_free_surfaces(void **data)
+{
+    struct i965_encoder_vp8_surface *vp8_surface;
+
+    if (!data || *data)
+        return;
+
+    vp8_surface = *data;
+
+    if (vp8_surface->scaled_4x_surface_obj) {
+        i965_DestroySurfaces(vp8_surface->ctx, &vp8_surface->scaled_4x_surface_id, 1);
+        vp8_surface->scaled_4x_surface_id = VA_INVALID_SURFACE;
+        vp8_surface->scaled_4x_surface_obj = NULL;
+    }
+
+    if (vp8_surface->scaled_16x_surface_obj) {
+        i965_DestroySurfaces(vp8_surface->ctx, &vp8_surface->scaled_16x_surface_id, 1);
+        vp8_surface->scaled_16x_surface_id = VA_INVALID_SURFACE;
+        vp8_surface->scaled_16x_surface_obj = NULL;
+    }
+}
+
+static void
+i965_encoder_vp8_allocate_surfaces(VADriverContextP ctx,
+                                   struct intel_encoder_context *encoder_context,
+                                   struct object_surface *obj_surface)
+{
+    struct i965_driver_data *i965 = i965_driver_data(ctx);
+    struct i965_encoder_vp8_context *vp8_context = encoder_context->vme_context;
+    struct i965_encoder_vp8_surface *vp8_surface;
+    int down_scaled_width_4x, down_scaled_height_4x;
+    int down_scaled_width_16x, down_scaled_height_16x;
+
+    if (!obj_surface || obj_surface->private_data)
+        return;
+
+    vp8_surface = calloc(1, sizeof(struct i965_encoder_vp8_surface));
+    vp8_surface->ctx = ctx;
+
+    down_scaled_width_4x = vp8_context->down_scaled_width_4x;
+    down_scaled_height_4x = vp8_context->down_scaled_height_4x;
+    i965_CreateSurfaces(ctx,
+                        down_scaled_width_4x,
+                        down_scaled_height_4x,
+                        VA_RT_FORMAT_YUV420,
+                        1,
+                        &vp8_surface->scaled_4x_surface_id);
+    vp8_surface->scaled_4x_surface_obj = SURFACE(vp8_surface->scaled_4x_surface_id);
+    i965_check_alloc_surface_bo(ctx, vp8_surface->scaled_4x_surface_obj, 1,
+                                VA_FOURCC('N', 'V', '1', '2'), SUBSAMPLE_YUV420);
+
+    down_scaled_width_16x = vp8_context->down_scaled_width_16x;
+    down_scaled_height_16x = vp8_context->down_scaled_height_16x;
+    i965_CreateSurfaces(ctx,
+                        down_scaled_width_16x,
+                        down_scaled_height_16x,
+                        VA_RT_FORMAT_YUV420,
+                        1,
+                        &vp8_surface->scaled_16x_surface_id);
+    vp8_surface->scaled_16x_surface_obj = SURFACE(vp8_surface->scaled_16x_surface_id);
+    i965_check_alloc_surface_bo(ctx, vp8_surface->scaled_16x_surface_obj, 1,
+                                VA_FOURCC('N', 'V', '1', '2'), SUBSAMPLE_YUV420);
+
+
+    obj_surface->private_data = vp8_surface;
+    obj_surface->free_private_data = i965_encoder_vp8_free_surfaces;
```

Comment:
Are you trying to avoid this @yakuizhao if they are already mapped by adding the conditional?